### PR TITLE
Make sure the current proxy is injected into repositories.

### LINKED ISSUE #2905: Proxy settings have no effect when analyzing certain maven poms
I am running ort in a corporate network with the following settings:

```
________ _____________________
\_____  \\______   \__    ___/ the OSS Review Toolkit, version 0.1.0-SNAPSHOT.
 /   |   \|       _/ |    |    Running 'analyze' under Java 11.0.6 on Linux with
/    |    \    |   \ |    |    ORT_DATA_DIR = /var/lib/jenkins/workspace/ort-pipeline@tmp/.ort
\_______  /____|_  / |____|    http_proxy = http://172.17.0.1:3128
        \/       \/
More environment variables:
https_proxy = http://172.17.0.1:3128
JAVA_HOME = /opt/java/openjdk
ANDROID_HOME = /opt/android-sdk
GOPATH = /go

```
The project to analyze is [Apache Commons Configuration](https://commons.apache.org/proper/commons-configuration/) 2.7. This is not a multi-module project, but it refers to an external parent pom (commons parent, which is shared by all Apache Commons components).

When analyzing the pom the proxy settings are obviously not effective; resolution of the parent pom fails with an error message as below:

```
org.apache.maven.project.ProjectBuildingException: Some problems were encountered while processing the POMs:
[FATAL] Non-resolvable parent POM for org.apache.commons:commons-configuration2:2.7: Could not transfer artifact org.apache.commons:commons-parent:pom:50 from/to central (https://repo.maven.apache.org/maven2): repo.maven.apache.org: Name or service not known and 'parent.relativePath' points at wrong local POM @ line 22, column 11
	at org.apache.maven.project.DefaultProjectBuilder.build(DefaultProjectBuilder.java:397)
	at org.ossreviewtoolkit.analyzer.managers.Maven.beforeResolution(Maven.kt:102)
	at org.ossreviewtoolkit.analyzer.PackageManager.resolveDependencies(PackageManager.kt:224)
	at org.ossreviewtoolkit.analyzer.Analyzer$analyzeInParallel$$inlined$use$lambda$1$1$1.invokeSuspend(Analyzer.kt:141)
	at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
	at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:56)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
```

### PR DESCRIPTION
**Pull request by oheger-bosch**
Resolves #2905.

Maven seems to fully initialize repositories only when they are read
from settings.xml or from a project pom. In a scenario where no
repositories are specified, the default repository - Maven Central -
is not assigned a proxy, even if the repository system session has a
valid proxy configuration.

To make sure that proxy settings are correctly applied, when creating
a request to execute Maven, the proxy is now explicitly injected into
all the repositories used by the request.

More details about the exact constellation addressed by this PR can be found in the referenced ticket.

### CODE REVIEW

File: analyzer/src/main/kotlin/managers/utils/MavenSupport.kt [hunk 1]
```diff
@@ -580,6 +592,11 @@ class MavenSupport(workspaceReader: WorkspaceReader) {
         )
     }
 
+    /**
+     * Convenience function to obtain the [MavenRepositorySystem] from the container.
+     */
+    private fun fetchMavenRepositorySystem() = container.lookup(MavenRepositorySystem::class.java, "default")
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/utils/MavenSupport.kt:598]: How about generalizing this like

    private inline fun <reified T> lookup(hint: String = "default") = container.lookup(T::class.java, hint)

and making use of it in *all* `container.lookup()` calls like

    val populator = lookup<MavenExecutionRequestPopulator>()

Ideally, that code simplification would go into a separate first commit as it's not strictly related to the actual fix.

Comment (oheger-bosch) [reply analyzer/src/main/kotlin/managers/utils/MavenSupport.kt]: Great idea. I implemented the generic lookup function. I made it public, however, as the Maven class also makes a lookup.

**Comment by sschuberth** [comment]
Something is wrong with your author / committer / signoff email, it should be in all cases your vanilla Bosch.IO address.

File: analyzer/src/main/kotlin/managers/Maven.kt [hunk 2]
```diff
@@ -96,7 +96,7 @@ class Maven(
     fun enableSbtMode() = also { sbtMode = true }
 
     override fun beforeResolution(definitionFiles: List<File>) {
-        val projectBuilder = mvn.container.lookup(ProjectBuilder::class.java, "default")
+        val projectBuilder = mvn.containerLookup<ProjectBuilder>()
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/Maven.kt:99]: Commit message: Please use [imperative mood](https://github.com/oss-review-toolkit/ort/blob/master/CONTRIBUTING.md#git-history), like

    Simplify the access to objects in the Maven Plexus container

(and also no dot at the end of the title).

File: analyzer/src/main/kotlin/managers/utils/MavenSupport.kt [hunk 3]
```diff
@@ -262,6 +262,7 @@ class MavenSupport(workspaceReader: WorkspaceReader) {
 
         populator.populateFromSettings(request, settings)
         populator.populateDefaults(request)
+        repositorySystemSession.injectProxy(request)
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/utils/MavenSupport.kt:265]: Commit message: Please remove the dot from the end of the commit message title, and move the `Resolves #2905.` part to just before the `Signed-off-by` footer.

**Pull request merged**